File: src/main/java/nu/fgv/register/server/spex/SpexCategoryRepository.java
package nu.fgv.register.server.spex;

import org.springframework.data.domain.Sort;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.querydsl.QuerydslPredicateExecutor;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.List;

@Repository
public interface SpexCategoryRepository extends JpaRepository<SpexCategory, Long>, QuerydslPredicateExecutor<SpexCategory> {

    @Query("""
              SELECT s FROM SpexCategory s
              WHERE s.id IN :ids
            """)
    List<SpexCategory> findByIds(@Param("ids") List<Long> ids, Sort sort);
}

File: src/main/kotlin/nu/fgv/register/server/repository/SpexCategoryRepository.kt
package nu.fgv.register.server.repository

import nu.fgv.register.server.domain.SpexCategory
import org.springframework.data.repository.PagingAndSortingRepository
import org.springframework.data.rest.core.annotation.RepositoryRestResource

@RepositoryRestResource
interface SpexCategoryRepository : PagingAndSortingRepository<SpexCategory, Long>
